query/executor_test.go

```diff
@@ -478,6 +487,66 @@ func TestQueryExecutor_Panic(t *testing.T) {
 	}
 }
 
+const goodStatement = `SELECT count(value) FROM cpu`
+
+func TestQueryExecutor_NotExecuted(t *testing.T) {
+	var executorFailIndex int
+	var executorCallCount int
+	queryStatements := []string{goodStatement, goodStatement, goodStatement, goodStatement, goodStatement}
+	queryStr := strings.Join(queryStatements, ";")
+	var closing chan struct{}
+
+	q, err := influxql.ParseQuery(queryStr)
+	if err != nil {
+		t.Fatalf("parsing %s: %v", queryStr, err)
+	}
+
+	e := NewQueryExecutor()
+	e.StatementExecutor = &StatementExecutor{
+		ExecuteStatementFn: func(stmt influxql.Statement, ctx *query.ExecutionContext) error {
+			defer func() { executorCallCount++ }()
+			if executorFailIndex == executorCallCount {
+				closing <- struct{}{}
+				close(closing)
+				select {
+				case <-ctx.Done():
+					return nil
+				}
+			} else {
+				return ctx.Send(&query.Result{Err: nil})
+			}
+		},
+	}
+	testFn := func(testName string, i int, failIndex *int) {
+		*failIndex = i
+		executorCallCount = 0
+		results := e.ExecuteQuery(q, query.ExecutionOptions{}, closing)
+
+		checkNotExecutedResults(t, results, testName, *failIndex, len(q.Statements))
+	}
+	for i := 0; i < len(q.Statements); i++ {
+		closing = make(chan struct{})
+		testFn("executor", i, &executorFailIndex)
+	}
+}
+
+func checkNotExecutedResults(t *testing.T, results <-chan *query.Result, testName string, failIndex int, lenQuery int) {
+	r := 0
+	cnt := 0
+	for result := range results {
+		if result.Err == query.ErrNotExecuted {
+			cnt++
+			if result.StatementID <= failIndex {
+				t.Fatalf("StatementID for ErrNotExecuted is wrong index: %d", result.StatementID)
```

Comment:
Should we just check that all the indexes we expect to fail are actually failing (build a list of fail indexes and compare to the expected list?) Here we could be missing if one index shows up twice and another is missing.